deploy-service/common/src/main/java/com/pinterest/deployservice/dao/HostDAO.java

```diff
@@ -72,4 +72,6 @@
     Collection<String> getNewHostIdsByGroup(String groupName) throws Exception;
 
     Collection<String> getFailedHostIdsByGroup(String groupName) throws Exception;
+
+    Collection<String> getExcludedHostsFromEnvByEnvId(String envId) throws Exception;
```

Comment:
I wonder if we should follow the pattern how we do deploy. Which means we rely on deployagent report groups instead of changing the logic here.